Exercises/.cmake/FindKokkos.cmake

```diff
@@ -0,0 +1,44 @@
+# Early return if Kokkos is already set up
+# We do not use Kokkos_FOUND as it is not globally defined
+if (TARGET Kokkos::kokkos)
+    return()
+endif ()
+
+set(SPACK_CXX $ENV{SPACK_CXX})
+if (SPACK_CXX)
+    message("found spack compiler ${SPACK_CXX}")
+    set(CMAKE_CXX_COMPILER ${SPACK_CXX} CACHE STRING "the C++ compiler" FORCE)
+    set(ENV{CXX} ${SPACK_CXX})
+endif ()
+
+if (NOT CMAKE_BUILD_TYPE)
+    set(default_build_type "RelWithDebInfo")
+    message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+    set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING
+            "Choose the type of build, options are: Debug, Release, RelWithDebInfo and MinSizeRel."
+            FORCE)
+endif ()
+
+set(Kokkos_COMMON_SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/dep/Kokkos)
+
+if (NOT KokkosTutorials_FORCE_INTERNAL_Kokkos)
+    find_package(Kokkos CONFIG)
+endif ()
+
+if (Kokkos_FOUND)
+    message(STATUS "Found Kokkos: ${Kokkos_DIR} (version \"${Kokkos_VERSION}\")")
+else ()
+    if (EXISTS ${Kokkos_COMMON_SOURCE_DIR})
+        add_subdirectory(${Kokkos_COMMON_SOURCE_DIR} Kokkos)
+    else ()
+        include(FetchContent)
+        FetchContent_Declare(
+                Kokkos
+                GIT_REPOSITORY https://github.com/kokkos/kokkos.git
+                GIT_TAG 4.0.01
```

Comment:
Why not the latest release? Did you intend to update in a separate pull request?

Reply:
Yes, I thought a separate PR made more sense.